test/test_settings.py

```diff
@@ -367,104 +367,90 @@ class TestSettingsEquality(object):
     A class defining tests for the standard implementation of == and != .
     """
 
-    def an_instance(self):
-        """
-        Return an instance of the class under test.  Each call to this method
-        must return a different object.  All objects returned must be equal to
-        each other.
-        """
-        overrides = {
-            h2.settings.SettingCodes.HEADER_TABLE_SIZE: 0,
-            h2.settings.SettingCodes.MAX_FRAME_SIZE: 16384,
-            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: 4,
-            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2**16,
-        }
-        return h2.settings.Settings(client=True, initial_values=overrides)
-
-    def another_instance(self):
-        """
-        Return an instance of the class under test.  Each call to this method
-        must return a different object.  The objects must not be equal to the
-        objects returned by an_instance.  They may or may not be equal to
-        each other (they will not be compared against each other).
-        """
-        overrides = {
-            h2.settings.SettingCodes.HEADER_TABLE_SIZE: 8080,
-            h2.settings.SettingCodes.MAX_FRAME_SIZE: 16388,
-            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: 100,
-            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2**16,
-        }
-        return h2.settings.Settings(client=False, initial_values=overrides)
-
-    def test_identical_eq(self):
+    SettingsStrategy = strategies.builds(
+        h2.settings.Settings,
+        client=strategies.booleans(),
+        initial_values=strategies.dictionaries(
+            keys=strategies.sampled_from([
+                h2.settings.SettingCodes.HEADER_TABLE_SIZE,
+                h2.settings.SettingCodes.ENABLE_PUSH,
+                h2.settings.SettingCodes.INITIAL_WINDOW_SIZE,
+                h2.settings.SettingCodes.MAX_FRAME_SIZE,
+                h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS,
+                h2.settings.SettingCodes.MAX_FRAME_SIZE,
+                h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE,
+            ]),
+            values=strategies.integers(min_value=0))
+        ) 
+
+    @given(o=SettingsStrategy)
+    def test_identical_eq(self, o):
```

Comment:
Also, a minor thing: can we pick variable names that are a bit more descriptive please?  😄